tests/integration.rs

```diff
@@ -2359,6 +2360,30 @@ mod tests {
         }
     }
 
+    fn pty_read(mut pty: std::fs::File) -> Receiver<String> {
+        let (tx, rx) = mpsc::channel::<String>();
+        thread::spawn(move || loop {
+            thread::sleep(std::time::Duration::new(1, 0));
+            let mut buf = [0; 512];
+            let _ = pty.read(&mut buf).unwrap();
+            let output = std::str::from_utf8(&buf).unwrap().to_string();
+            tx.send(output).unwrap();
+        });
+        rx
+    }
+
+    fn get_pty_paths(api_socket: &str) -> PathBuf {
```

Comment:
Originally I had this function returning both the serial and console paths at once since I didn't think I'd be trying to get one without the other at that time. Little did I know X(.

Reply:
Done.